README.md

```diff
@@ -56,38 +72,38 @@ module "my-cluster" {
     }
   ]
 }
+
+resource "kubernetes_namespace" "example" {
+  metadata {
+    name = "example"
+  }
+}
 ```
+
 ## Conditional creation
 
 Sometimes you need to have a way to create EKS resources conditionally but Terraform does not allow to use `count` inside `module` block, so the solution is to specify argument `create_eks`.
 
-Using this feature _and_ having `manage_aws_auth=true` (the default) requires to set up the kubernetes provider in a way that allows the data sources to not exist.
-
+If you also want to sometimes create kubernetes resources then it gets slightly more complicated:
 ```hcl
-data "aws_eks_cluster" "cluster" {
-  count = var.create_eks ? 1 : 0
-  name  = module.eks.cluster_id
-}
-
 data "aws_eks_cluster_auth" "cluster" {
   count = var.create_eks ? 1 : 0
   name  = module.eks.cluster_id
 }
 
 # In case of not creating the cluster, this will be an incompletely configured, unused provider, which poses no problem.
 provider "kubernetes" {
-  host                   = element(concat(data.aws_eks_cluster.cluster[*].endpoint, list("")), 0)
-  cluster_ca_certificate = base64decode(element(concat(data.aws_eks_cluster.cluster[*].certificate_authority.0.data, list("")), 0))
+  host                   = module.eks.cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
   token                  = element(concat(data.aws_eks_cluster_auth.cluster[*].token, list("")), 0)
   load_config_file       = false
-  version                = "1.10"
 }
 
-# This cluster will not be created
+# This cluster will not be created when create_eks = false
 module "eks" {
   source = "terraform-aws-modules/eks/aws"
 
-  create_eks = false
+  create_eks = var.create_eks
```

Comment:
Where is `var.create_eks` coming from ? Why don't use `create_eks = false` directly ?

Reply:
From that level. The variable block needs adding. Or the whole section removing